service/controller/resource/monitoring/verticalpodautoscaler/resource.go

```diff
@@ -165,7 +165,23 @@ func (r *Resource) getMaxCPU(ctx context.Context) (*resource.Quantity, error) {
 		return nil, microerror.Mask(nodeCpuNotFoundError)
 	}
 
-	q, err := quantityMultiply(nodeCpu, 0.9)
+	n, ok := nodeCpu.AsInt64()
+	if !ok {
+		return nil, microerror.Maskf(quantityConversionError, nodeCpu.String())
+	}
+
+	var q *resource.Quantity
+
+	if n <= 8 {
```

Comment:
Still, what are those magic numbers? :D

Reply:
https://github.com/giantswarm/giantswarm/issues/25877#issuecomment-1437140886

Reply:
I will create a table later.

Reply:
Sure because I think this should be documented here TBH.

Also, I might be wrong on my hypothesis but I think that not allowing more than 50% of node CPU should work everywhere. I'd like to know why deu01 requires so much but I guess this is due to rule evaluation again?

Reply:
There might be a correlation about rule evaluation * nodes = CPU usage but we should be fine with 50% top

Reply:
I.e. I do not think the number of allocatable CPUs on the node should be a good way to know if we need to limit to 50% or more of CPU usage but it should be around the number of worker nodes.

We can revisit this later tbh. If 50 works, let's go with 50

Reply:
So, you mean 50% for all?

Reply:
But We have grizzly failing! It had to requests 2 CPU at least

Reply:
grizzly nodes have 4 cpu. so if we have 70% (2.8)

Reply:
I see 0.5 is enough, but for the nodes that has only 4 cpus and needs to evaluate something. 
I really want to avoid the eviction of Prometheus Pod circumstances, due to high cpu usage.

Reply:
Then maybe we set it to 70 everywhere 😅

Reply:
Yes, If you wish)

Reply:
done)

Reply:
Can nodes get evicted due to high CPU usage?
It's not like RAM where you can OOM. Your app just gets slow, has problems catching back, and maybe can't do its job anymore. But it won't crash (except unknown reasons like strange race conditions that only happen when under a lot of stress).